lib/sgov/sgov-adapter.ts

```diff
@@ -0,0 +1,227 @@
+import getSurroundingsComplexAttributes from "./queries/get-surroundings-complex-attributes.sparql";
+import getHierarchy from "./queries/get-hierarchy.sparql";
+import getGroup from "./queries/get-group.sparql";
+import {CimAdapter, IriProvider} from "../cim";
+import {SparqlQueryRdfSource} from "../io/rdf/sparql/sparql-query-rdf-source";
+import {HttpFetch} from "../io/fetch/fetch-api";
+import {RdfSource, RdfSourceWrap} from "../core/adapter/rdf";
+import {isSgovClass, loadSgovClass} from "./entity-adapters/sgov-class-adapter";
+import {CoreResource, ReadOnlyMemoryStore} from "../core";
+import {FederatedSource} from "../io/rdf/federated/federated-rdf-source";
+import {RDFS, SKOS} from "./sgov-vocabulary";
+import {isSgovAssociation, loadSgovAssociation} from "./entity-adapters/sgov-association-adapter";
+import {isSgovAttribute, loadSgovAttribute} from "./entity-adapters/sgov-attribute-adapter";
+import {PimClass} from "../pim/model";
+
+const getSurroundings = [
+  getSurroundingsAttributes,
+  getSurroundingsInwardsRelations,
+  getSurroundingsOutwardsRelations,
+  getSurroundingsParent,
+  getSurroundingsComplexAttributes,
+];
+
+const compressQueryString = (query: string): string => {
+  return query.replace(/^\s*#.*$/mg, "").replace(/[ \t]+/g, " ");
+};
+
+const searchQuery = (searchString: string) => compressQueryString(search.replace("%QUERY%", `"${jsStringEscape(searchString)}"`));
+const getClassQuery = (cimIri: string) => compressQueryString(getClass.replace(/%NODE%/g, `<${cimIri}>`));
+const getSurroundingsQueries = (cimIri: string) => getSurroundings.map(q => compressQueryString(q.replace(/%NODE%/g, `<${cimIri}>`)));
+const getHierarchyQuery = (cimIri: string) => compressQueryString(getHierarchy.replace(/%NODE%/g, `<${cimIri}>`));
+const getGroupQuery = (cimIri: string) => compressQueryString(getGroup.replace(/%NODE%/g, `<${cimIri}>`));
+
+const IRI_REGEXP = new RegExp("^https?://");
+
+/**
+ * Adapter for SGOV (Semantic Government Vocabulary) CIM type.
+ *  - https://slovník.gov.cz/sparql
+ */
+export class SgovAdapter implements CimAdapter {
+  protected readonly endpoint: string;
+  protected readonly httpFetch: HttpFetch;
+  protected iriProvider: IriProvider;
+
+  protected resourceGroupCache = new Map<string, string[] | undefined>();
+
+  constructor(endpoint: string, httpFetch: HttpFetch) {
+    this.endpoint = endpoint;
+    this.httpFetch = httpFetch;
+  }
+
+  setIriProvider(iriProvider: IriProvider): void {
+    this.iriProvider = iriProvider;
+  }
+
+  async search(searchString: string): Promise<PimClass[]> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, searchQuery(searchString));
+    await source.query();
+
+    const results = await source.property("__search_results", "__has_search_result");
+
+    let sorted = [];
+    for (const result of results) {
```

Comment:
This would be a nice method, it would also tackle the issue wherein some part of the code you have unsorted data in the ```sorted`` variable. In addition it is not really what is the content of sorted besides that fact that is should be sorted, which is not in the next 7 lines or so.